typing/typedtree.ml

```diff
@@ -25,8 +25,18 @@ type partial = Partial | Total
 type attribute = Parsetree.attribute
 type attributes = attribute list
 
-type pattern =
-  { pat_desc: pattern_desc;
+type value = Value_pattern
```

Comment:
That name seems a bit too generic, don't you agree?

Why don't you mint a `Pattern` submodule to put it (and all the following types) in.

Reply:
I thought about it but I worried that it would be out-of-style with the rest. (Contrarily to our own parmatch/matching, the structure here affects every consumer of the typedtree, which is a lot of people.) I'm willing to go that route if there is buy-in from the other reviewers.